### PR TITLE
Jared/layer vs rewrite

### PR DESCRIPTION
**Pull request by jared-2016**
@JamesMBallard Ready for review. Thanks!

### CODE REVIEW

File: ArcGISRuntimeSDKQt_QMLSamples/Maps/DisplayLayerViewDrawState/README.md [hunk 1]
```diff
@@ -6,29 +6,44 @@ Determine if a layer is currently being viewed.
...
+* `LOADING`: The layer in the view is loading. Once loading has completed, the layer will be available for display in the view. If there was a problem loading the layer, the status will be set to ERROR.
+* `ERROR`: The layer in the view has an unrecoverable error. When the status is `ERROR`, the layer cannot be rendered in the view. For example, it may have failed to load, be an unsupported layer type, or contain invalid data.
+* `WARNING`: The layer in the view has a non-breaking problem with its display, such as incomplete information (eg. by requesting more features than the max feature count of a service) or a network request failure.
+
+If your device supports airplane mode, you can toggle this on and pan around the map to see layers display the WARNING status when they cannot online fetch data. Toggle airplane mode back off to see the warning disappear.
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_QMLSamples/Maps/DisplayLayerViewDrawState/README.md:45]:
```suggestion
If your device supports airplane mode, you can toggle this on and pan around the map to see layers display the Enums.LayerViewStatusWarning status when they cannot online fetch data. Toggle airplane mode back off to see the warning disappear.
```

File: ArcGISRuntimeSDKQt_QMLSamples/Maps/DisplayLayerViewDrawState/DisplayLayerViewDrawState.qml [hunk 2]
```diff
@@ -15,134 +15,171 @@
...
-            const index = getindex(layer);
-            // get Current Status
-            const status = viewStatusString(layerViewState);
-            // change name if layer loaded
-            layerViewModel.setProperty(index, "name", layer.name);
-            // update Status in ListModel
-            layerViewModel.setProperty(index, "status", status);
-        }
+            // only update list if the layer is the feature layer.
+            if (layer != featureLayer)
```
[the remark this answers is not in the document]
Comment (jared-2016) [reply ArcGISRuntimeSDKQt_QMLSamples/Maps/DisplayLayerViewDrawState/DisplayLayerViewDrawState.qml:79]: I could compare names easily enough. Good catch!

File: ArcGISRuntimeSDKQt_QMLSamples/Maps/DisplayLayerViewDrawState/Info.plist [hunk 3]
```diff
@@ -48,7 +48,7 @@
 		<string>UIInterfaceOrientationLandscapeLeft</string>
 		<string>UIInterfaceOrientationLandscapeRight</string>
 	</array>
-    <key>UILaunchStoryboardName</key>
-    <string>LaunchScreen</string>
+        <key>UILaunchStoryboardName</key>
+        <string>LaunchScreen</string>
```
[the remark this answers is not in the document]
Comment (jared-2016) [reply ArcGISRuntimeSDKQt_QMLSamples/Maps/DisplayLayerViewDrawState/Info.plist:52]: Looking [here](https://github.com/Esri/arcgis-runtime-samples-qt/blob/jared/layer_vs_rewrite/ArcGISRuntimeSDKQt_QMLSamples/Maps/DisplayLayerViewDrawState/Info.plist), seems fine to me unless im missing something?

**Comment by jared-2016** [comment]
@JamesMBallard Ready for your eyes again. I think I got them all and the others I responded too and will make the change if you deem necessary. Thanks!

File: ArcGISRuntimeSDKQt_QMLSamples/Maps/DisplayLayerViewDrawState/Info.plist (hunk 3 above)
[the remark this answers is not in the document]
Comment (JamesMBallard) [reply ArcGISRuntimeSDKQt_QMLSamples/Maps/DisplayLayerViewDrawState/Info.plist:52]: It looks off in the diff. It looks fine when I view the entire file. No changes needed.

File: ArcGISRuntimeSDKQt_CppSamples/Maps/DisplayLayerViewDrawState/arcgisruntime.pri [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 #-------------------------------------------------
 
 contains(QMAKE_HOST.os, Windows):{
-  iniPath = $$(ALLUSERSPROFILE)\\EsriRuntimeQt\\ArcGIS Runtime SDK for Qt $${ARCGIS_RUNTIME_VERSION}.ini
+  iniPath = $$(ALLUSERSPROFILE)\EsriRuntimeQt\ArcGIS Runtime SDK for Qt $${ARCGIS_RUNTIME_VERSION}.ini
```
[the remark this answers is not in the document]
Comment (JamesMBallard) [reply ArcGISRuntimeSDKQt_CppSamples/Maps/DisplayLayerViewDrawState/arcgisruntime.pri:17]: I see a few examples that are single backslash. This should be fine.

**Pull request merged**